packages/backend/src/middleware/tenant/index.ts

```diff
@@ -0,0 +1,58 @@
+import { ApolloContext, TenantedApolloContext } from '../../app'
+
+type Request = () => Promise<unknown>
+
+interface TenantValidateMiddlewareArgs {
+  deps: { ctx: ApolloContext }
+  tenantIdInput: string | undefined
+  onFailValidation: Request
+  next: Request
+}
+
+export async function validateTenantMiddleware(
```

Comment:
Validate for any `TenantedApolloContext` and `tenantIdInput` is present.

Reply:
@mkurapov @njlie @BlairCurrey 
This is a bit of a hack, where the `forTenantId` is sitting on the `TenantedApolloContext`. Will figure that one out.

But the rest basically does what we talked about, without having to update any additional code on each of the resolvers. 

As long as the resolver type is `TenantedApolloContext`, the `tenantId` validations are performed, and it is safe to always use `forTenantId`.

What do you guys think?

Reply:
I think this is a bit simpler:

```ts
export async function setForTenantIdMiddleware(
  args: SetForTenantIdMiddlewareArgs
): ReturnType<Request> {
  const {
    deps: { ctx },
    tenantIdInput,
    onFail,
    next
  } = args
  if (!('tenant' in ctx && 'isOperator' in ctx)) return onFail()
  if (!ctx.isOperator && tenantIdInput) return onFail()

  ctx.forTenantId = tenantIdInput ?? ctx.tenant.id
  return next()
}
```

Differences:

- Errors when we dont have the tenant nor operator. Correct me if I'm wrong on this but we never want to allow an request without the tenant and if they are an operator. Hence `tenant` and `isOperator` are required properties on the `TenantedApolloContext`
- Some renaming to convey this is just getting the id, not so much validating

Reply:
> This is a bit of a hack, where the forTenantId is sitting on the TenantedApolloContext. Will figure that one out.

For the types I think you can remove it from the initial `TenantedApolloContext` and make a new interface like:

```ts
interface ForTenantIdContext extends TenantedApolloContext {
  forTenantId: string
}
```

And use in the middleware like:

```ts
export function setForTenantIdGraphQLMutationMiddleware(): {
  Mutation: IMiddleware<TenantedApolloContext, ForTenantIdContext>
} {
```

Then update the context type passed into the resolvers: 


```ts
export const getWalletAddress: QueryResolvers<ForTenantIdContext>['walletAddress'] =
```